.github/workflows/run_tests.yml

```diff
@@ -0,0 +1,58 @@
+name: Run Unit Tests
+
+# Remove this for now, circleci is getting in the way with tests failing
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  # trigger on pull requests
+  pull_request:
+
+  # trigger on all commits to master and tagged commits
+  push:
+    branches:
+      - "master"
+
+  # trigger on request
+  workflow_dispatch:
+
+jobs:
+  run_tests:
+    name: Run tests on ${{ matrix.os }} with python ${{ matrix.pyver-env.pyver }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macOS-latest, ubuntu-latest, windows-latest]
+        pyver-env: [ {pyver: '3.6', env: 'test-oldest_env.yaml'},
+                     {pyver: '3.7', env: 'test-compatible_env.yaml'},
+                     {pyver: '3.8', env: 'test_env.yaml'},
+                     {pyver: '3.9', env: 'test_env.yaml'},
+                     {pyver: '3.10', env: 'test_env.yaml'} ]
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2.3.4
+        with:
+          submodules: true
+
+      - name: Create Python Environment
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          python-version: ${{ matrix.pyver-env.pyver }}
+          environment-file: .github/workflows/conda-envs/${{ matrix.pyver-env.env }}
+          channels: conda-forge,defaults
```

Comment:
I would remove `defaults` from all conda channels. We should pull exclusively from `conda-forge` to avoid incompatibilities.

Reply:
I'd recommend using mambaforge here: https://github.com/marketplace/actions/setup-miniconda#example-10-miniforge

Reply:
What's the benefit of using `Mambaforge` here instead of just setting channels to conda-forge exclusively?